src/devices/CharacterLcd/LcdConsole.cs

```diff
@@ -0,0 +1,537 @@
+                {
+                    SetCursorPosition(0, CursorTop + 1);
+                }
+                else if (LineFeedMode.HasFlag(LineFeedMode.ScrollUp))
+                {
+                    ScrollUp();
+                    SetCursorPosition(0, Size.Height - 1); // Go to beginning of last line
+                }
+                else
+                {
+                    CursorLeft = Size.Width; // We are outside the screen, any further writes will be ignored
+                }
+            }
+        }
+
+        /// <summary>
+        /// Refreshes the display from the cache (i.e. after a scroll operation)
+        /// Does not change the cursor position
+        /// </summary>
+        private void RefreshFromBuffer()
+        {
+            int x = CursorLeft;
+            int y = CursorTop;
+            for (int i = 0; i < Size.Height; i++)
+            {
+                _lcd.SetCursorPosition(0, i);
+                byte[] buffer = MapChars(_currentData[i].ToString());
+                _lcd.Write(buffer);
+            }
+            SetCursorPosition(x, y);
+        }
+
+        /// <summary>
+        /// Scrolls the text up by one row, clearing the last row. Does not change the cursor position.
+        /// </summary>
+        private void ScrollUp()
+        {
+            for (int i = 1; i < Size.Height; i++)
+            {
+                _currentData[i - 1] = _currentData[i];
+            }
+            _currentData[Size.Height - 1] = new StringBuilder(new String(' ', Size.Width));
+            RefreshFromBuffer();
+        }
+
+        /// <summary>
+        /// This is expected to be called only with a string length of less or equal the remaining number of characters on the current line. 
+        /// </summary>
+        private void WriteCurrentLine(string line)
+        {
+            // Replace the existing chars at the given position with the new text
+            _currentData[CursorTop].Remove(CursorLeft, line.Length);
+            _currentData[CursorTop].Insert(CursorLeft, line);
+            byte[] buffer = MapChars(line);
+            _lcd.Write(buffer);
+            CursorLeft += line.Length;
+        }
+
+        /// <summary>
+        /// Loads the character set for the given culture, if available
+        /// </summary>
+        /// <param name="culture">Culture to load</param>
+        /// <param name="unknownCharacter">The character to print for unknown letters, default: ?</param>
+        /// <param name="factory">Character encoding factory that delivers the mapping of the Char type to the hardware ROM character codes. May add special characters into 
+        /// the character ROM. Default: Null (Use internal factory)</param>
+        /// <returns>True if all important characters for that culture are available and loaded.</returns>
+        public bool LoadCulture(CultureInfo culture, char unknownCharacter = '?', LcdCharacterEncodingFactory factory = null)
```

Comment:
I think this should just take Encoding directly and so should method name be called (at least not seeing culture being used anywhere)

Reply:
See a few lines down. LcdCharacterEncodingFactory.Create() takes the culture to obtain the language name.

Reply:
The factory parameter is optional and allows specifying user-defined custom mappings.

Reply:
Separated `LoadEncoding` from `CreateEncoding` now.